CONTRIBUTING.md

```diff
@@ -0,0 +1,124 @@
+section if you are contributing to this repository.
+
+If you submit a pull request that resolves an open issue, please help us to keep our issue list small by adding
+`closes: #{$ISSUE_NO}` to your commit message. GitHub will use this tag to auto-close the issue if your PR is merged.
+
+#### What is an Algorithm?
+
+An Algorithm is one or more functions (or classes) that:
+
+- Take one or more inputs.
+- Perform some internal calculations or data manipulations.
+- Return one or more outputs.
+- Have minimal side effects.
+
+Algorithms should be packaged in a way that would make it easy for readers to put them into larger programs.
+
+Algorithms should:
+
+- Have intuitive class and function names that make their purpose clear to readers.
+- Use TypeScript naming conventions and intuitive variable names to ease comprehension.
+- Be flexible to take different input values.
+- Raise TypeScript exceptions (RangeError, etc.) on erroneous input values.
+
+Algorithms in this repo should not be how-to examples for existing TypeScript packages. Instead, they should perform
+internal calculations or manipulations to convert input values into different output values. Those calculations or
+manipulations can use data types, classes, or functions of existing TypeScript packages but each algorithm in this repo
+should add unique value.
+
+#### File Naming Convention
+
+- Filenames should use the UpperCamelCase (PascalCase) style.
+- There should be no spaces in filenames.
+- **Example:** `UserProfile.ts` is allowed. Do not use `userprofile.ts`, `Userprofile.ts`, `user-Profile.ts`, `userProfile.ts`; these naming conventions are discouraged.
+
+#### Commit Messages Formatting
+
+- Prefer to use the following format: `<type>: <short summary>`. If necessary, put any extra information in the description.
+- Commit types include (but are not limited to): 
+  - **docs**: Documentation only changes
+  - **feat**: A new feature
+  - **fix**: A bug fix
+  - **test**: Adding or fixing tests
+  - **chore**: CI / code quality / minor quality of life improvements
+
+- **Examples**: 
+  - `feat: add quicksort algorithm`
+  - `chore: fix spelling`
+  - `fix: improper error message`
+  - `docs: add contributing guidelines`
+  - `test: add test for quicksort algorithm`
+
+#### Module System
+
+We use the [ES Module](https://hacks.mozilla.org/2018/03/es-modules-a-cartoon-deep-dive/) system, which brings an official, standardized module system to TypeScript.
+
+It roughly means you will need to use `export` and `import` statements instead of `module.exports` and `require()`.
+
+#### Coding Style
+
+To maximize the readability and correctness of our code, we require that new submissions follow the
+[TypeScript Standard Style](https://github.com/standard/ts-standard).
+
+A few (but not all) of the things to keep in mind:
+
+- Use `lowerCamelCase` with the leading character as lowercase for identifier names (variables and functions).
```

Comment:
Add a point to use `UpperCamelCase` for class, type, and interface names.
And to use `PASCAL_CASE` for universal constants.